### PR TITLE
Data Conversion

### LINKED ISSUE #16: Data Conversion for Better Displaying
There are a couple things we would like to take care when displaying the data, for example:

1. read byte from the database, and then convert them to protobufs for displaying
2. read timestamp in long type, and convert them to human-readable timestamp type in PST 
3. read integer, and convert them to protobuf enums, and display human-readable enum values

### PR DESCRIPTION
**Pull request by wdcroslen**
Files Changes:
  database-tables.js      :     Lines 32 -47 ,  85- 145,
  table-style.css         : Line  15

- convert "proto"  (Json) to better viewable data
- read long timestamp, and convert to human-readable timestamp in PST
- read integer, and display human-readable enum values
Genre_EnumChanges
Millis ProtoChanges

### CODE REVIEW

**Review by gagomez — COMMENTED** [review]
Looks like this includes all the changes in the large commit? Can you share what has changed?

File: src/main/webapp/database-tables.js [hunk 1]
```diff
@@ -22,33 +28,34 @@ function showDatabase() {
...
-        }
 
         const colSchemas = tableData.columnSchemas;
+        var finalDataTable = tableData;
+    
+        for (var i =0; i<colSchemas.length; i++) {
+            if (colSchemas[i].columnName.endsWith('Millis')){
+                var finalDataTable = dataConversionMillis(i,finalDataTable);
+            } else if (colSchemas[i].columnName == "Genre") {
+                var finalDataTable = dataConversionEnum(i,finalDataTable);
```
Comment (jiaxinz-g) [inline src/main/webapp/database-tables.js]: Rename dataConversionEnum to dataConversionGenreEnum. 

Enum conversion without know what the enum is hard to generalize, we will probably need a map from the column name to the enums. For now, I think this looks great! Just try to specify that this works for Genre and will need to extend to other enums.

**Review by jiaxinz-g — APPROVED** [review]

File: src/main/webapp/database-tables.js (hunk 1 above)
Comment (wdcroslen) [reply src/main/webapp/database-tables.js]: Thank you, this has been changed!

File: src/main/webapp/database-tables.js [hunk 2]
```diff
@@ -33,15 +33,26 @@ function showDatabase() {
         const isEmpty = tableData.isEmpty;
 
         const colSchemas = tableData.columnSchemas;
+        var finalDataTable = tableData;
+    
+        for (var i =0; i<colSchemas.length; i++) {
```
Comment (sasymer) [inline src/main/webapp/database-tables.js:38]: nit: could rename i to index

File: src/main/webapp/database-tables.js [hunk 3]
```diff
@@ -68,6 +79,67 @@ function login() {
   });
 }
 
+//TODO: Pass another paramater to combine data conversion functions 
+function dataConversionMillis(column,tableData){
+    tableData.columnSchemas[column].schemaType = "TIMESTAMP";
+    for (var i =0; i<tableData.dataTable.length; i++) {
```
Comment (sasymer) [inline src/main/webapp/database-tables.js]: Same comment (maybe index instead of i but not a big deal)

File: src/main/webapp/database-tables.js [hunk 4]
```diff
@@ -68,6 +79,67 @@ function login() {
...
+function dataConversionProto(column,tableData) {
+    //TODO: work on clicking the { to open and close
+    for (var i = 0; i<tableData.dataTable.length; i++) {
+      var json = tableData.dataTable[i][column];
+      var newString = "";
+      var tab = Boolean(false);
+
+      for (var j = 0; j<json.length; j++) {
+          if (json[j] == ',' ) {
+              newString += (Boolean(tab) ? (json[j] + "\n"+"\t") : (json[j] + "\n"));
```
Comment (sasymer) [inline src/main/webapp/database-tables.js]: Could you make the "\n" + "\t" into "\n\t"?

**Review by sasymer — APPROVED** [review]
Looks good!

File: src/main/webapp/database-tables.js (hunk 4 above)
Comment (wdcroslen) [reply src/main/webapp/database-tables.js]: Done!

File: src/main/webapp/database-tables.js (hunk 3 above)
Comment (wdcroslen) [reply src/main/webapp/database-tables.js]: Its more clear so I changed it!

**Pull request merged**